core/modules/qdisp/Executive.cc

```diff
@@ -208,35 +229,26 @@ void Executive::markCompleted(int jobId, bool success) {
     }
 }
 
-void Executive::requestSquash(int jobId) {
-    JobQuery::Ptr toSquash;
-    {
-        std::lock_guard<std::mutex> lock(_jobsMutex);
-        auto iter = _jobMap.find(jobId);
-        if(iter == _jobMap.end()) {
-            LOGF_WARN("requestSquash invalid jobID %1%" % jobId);
-            return;
-        }
-        toSquash = iter->second;
-    }
-    // release mutex before calling cancel.
-    toSquash->cancel();
-}
-
 void Executive::squash() {
     bool alreadyCancelled = _cancelled.exchange(true);
     if(alreadyCancelled) {
-        LOGF_WARN("Executive::squash() already cancelled! refusing.");
+        LOGF_WARN("%1% Executive::squash() already cancelled! refusing." % _id);
         return;
     }
 
-    LOGF_INFO("Trying to cancel all queries...");
+    LOGF_WARN("%1% Executive::squash Trying to cancel all queries..." % _id);
+    std::deque<JobQuery::Ptr> jobsToCancel;
```

Comment:
deque is reasonably efficient, it's got a a very small code footprint, it's simple, and I'd be incredibly surprised if a profiler indicated this would be a good focus for optimization.